lib/tool_shed/galaxy_install/tools/data_manager.py

```diff
@@ -29,21 +30,18 @@ def data_manager_config_elems_to_xml_file(self, config_elems, config_filename):
         of config_filename.
         """
         data_managers_path = self.data_managers_path
-        lock = threading.Lock()
-        lock.acquire(True)
+        if data_managers_path:
+            root_str = '<?xml version="1.0"?><data_managers tool_path="%s"></data_managers>' % data_managers_path
+        else:
+            root_str = '<?xml version="1.0"?><data_managers></data_managers>'
+        root = ElementTree.fromstring(root_str)
+        for elem in config_elems:
+            root.append(elem)
         try:
             with RenamedTemporaryFile(config_filename, mode='w') as fh:
-                if data_managers_path is not None:
-                    fh.write('<?xml version="1.0"?>\n<data_managers tool_path="%s">\n    ' % data_managers_path)
-                else:
-                    fh.write('<?xml version="1.0"?>\n<data_managers>\n    ')
-                for elem in config_elems:
-                    fh.write(xml_util.xml_to_string(elem))
-                fh.write('</data_managers>\n')
-        except Exception:
-            log.exception("Exception in DataManagerHandler.data_manager_config_elems_to_xml_file")
-        finally:
-            lock.release()
+                fh.write(xml_to_string(root))
+        except Exception as e:
+            log.exception("Exception in DataManagerHandler.data_manager_config_elems_to_xml_file:\n %s", str(e))
```

Comment:
This change was not needed, `log.exception()` already adds the exception info to the log message.